general/summary/cli.go

```diff
@@ -0,0 +1,229 @@
+package summary
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils/commandsummary"
+	"github.com/jfrog/jfrog-cli/utils/cliutils"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	commonCliUtils "github.com/jfrog/jfrog-cli-core/v2/common/cliutils"
+	coreConfig "github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	securityUtils "github.com/jfrog/jfrog-cli-security/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/urfave/cli"
+)
+
+type MarkdownSection string
+
+const (
+	Security  MarkdownSection = "security"
+	BuildInfo MarkdownSection = "build-info"
+	Upload    MarkdownSection = "upload"
+)
+
+const (
+	JfrogCliSummaryDir = "jfrog-command-summary"
+	MarkdownFileName   = "markdown.md"
+)
+
+var markdownSections = []MarkdownSection{Security, BuildInfo, Upload}
+
+func (ms MarkdownSection) String() string {
+	return string(ms)
+}
+
+// Creates a final summary of recorded CLI commands that were executed on the current machine.
+// The summary is generated in Markdown format and saved in the root directory of JFROG_CLI_COMMAND_SUMMARY_OUTPUT_DIR.
+func GenerateSummaryMarkdown(c *cli.Context) error {
+	if !ShouldGenerateSummary() {
+		return fmt.Errorf("cannot generate command summary: the output directory for command recording is not defined. "+
+			"Please set the environment variable %s before executing your commands to view their summary", coreutils.OutputDirPathEnv)
+	}
+	// Get URL and Version to generate summary links
+	serverUrl, majorVersion, err := extractServerUrlAndVersion(c)
+	if err != nil {
+		log.Warn("Failed to get server URL or major version: %v. This means markdown URLs will be invalid!", err)
```

Comment:
Why are we not returning the error, and just adding a warning message?
We need a very good reason to ignore error messages. Let's discuss this.